pages/volunteer/ProfileCarouselDirective.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppiaFoundationWebsite.directive('profileCarousel', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      desktopCarousel: '@desktop',
+      mobileCarousel: '@mobile',
+      activeTab: '@tab'
+    },
+    templateUrl: '/pages/volunteer/profileCarouselTemplate.html',
+    controller: ['$scope', 'VolunteerProfilesService',
+      function($scope, VolunteerProfilesService) {
+        $scope.slideInterval = 0;
+        $scope.noWrapSlides = false;
+        var DESIGN_TAB = 'Design';
+        var DEVELOPMENT_TAB = 'Development';
+        var RESEARCH_TAB = 'Research';
+        var MARKETING_TAB = 'Marketing';
+        var designSlide = {
+          active: true
```

Comment:
Hm, this is a bit of a strange way to code it, since the value of "active" never changes. So it seems at first glance like all four tabs are active.

Instead i think it would probably be more natural to have $scope.activeTab = ACTIVE_TAB_DESIGN etc. where ACTIVE_TAB_DESIGN etc. are named constants.

Reply:
I think I misunderstood [UIBootstrap's Carousel doc](https://angular-ui.github.io/bootstrap/#!#carousel) here. It is supposed to be the index of the active slide. I was afraid one slide cycling would trigger the other slide to cycle too. I will be more careful with this. Adding comment too!.